src/LuaInterface.cpp

```diff
@@ -560,42 +560,51 @@ void LuaInterface::renameVar(TVar* var)
     //this assumes anything like reparenting has been done
     L = interpreter->pGlobalLua;
     QList<TVar*> vars = varOrder(var);
-    QString oldVariableDeleteCode = vars.at(0)->getName();
+    QString oldVariable = vars.at(0)->getName();
     QString newName;
     if (vars.size() > 1) {
         newName = vars[0]->getName();
     }
     for (int i = 1; i < vars.size(); i++) {
         int kType = vars[i]->getKeyType();
         if (kType == LUA_TNUMBER) {
-            oldVariableDeleteCode.append(QStringLiteral("[%1]").arg(vars.at(i)->getName()));
+            oldVariable.append(QStringLiteral("[%1]").arg(vars.at(i)->getName()));
             if (i < vars.size() - 1) {
                 newName.append("[" + vars[i]->getName() + "]");
             }
         } else if (kType == LUA_TTABLE) {
             renameCVar(vars);
             return;
         } else {
-            oldVariableDeleteCode.append(QStringLiteral("[\"%1\"]").arg(vars.at(i)->getName()));
+            oldVariable.append(QStringLiteral("[\"%1\"]").arg(vars.at(i)->getName()));
             if (i < vars.size() - 1) {
                 newName.append(QStringLiteral("[\"%1\"]").arg(vars.at(i)->getName()));
             }
         }
     }
-    if (var->getNewKeyType() == LUA_TNUMBER) {
-        newName.append(QStringLiteral("[%1]").arg(vars.back()->getNewName()));
+
+    if (vars.size() <= 1) {
```

Comment:
This is the only interesting bit, please do not review the other code, that was just renaming variables so they make more sense!